tests/unit/prometheus_http_test.cc

```diff
@@ -0,0 +1,92 @@
+/*
+ * This file is open source software, licensed to you under the terms
+ * of the Apache License, Version 2.0 (the "License").  See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership.  You may not use this file except in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+/*
+ * Copyright 2024 ScyllaDB
+ */
+
+#include "loopback_socket.hh"
+
+#include <seastar/core/metrics.hh>
+#include <seastar/core/prometheus.hh>
+#include <seastar/http/common.hh>
+#include <seastar/testing/test_case.hh>
+#include <seastar/testing/thread_test_case.hh>
+
+#include <boost/test/tools/old/interface.hpp>
+
+using namespace seastar;
+using namespace httpd;
+
+namespace {
+
+struct test_metrics {
+    metrics::metric_groups _metrics;
+
+    void setup_metrics() {
+        _metrics.add_group("aaaa", {
+            metrics::make_gauge("int_test", [] { return 10; }, metrics::description{"simple minimal test"}),
+            metrics::make_gauge("double_test", [] { return 1234567654321.0; }, metrics::description{"test that a long double is printed fully and not in scientific notation"}),
+            metrics::make_counter("counter_test", [] () -> int64_t { return 1234567654321; }, metrics::description{"test with a long counter value"}),
+        });
+    }
+};
+
+future<> test_prometheus_metrics_body() {
+    test_metrics metrics;
+    metrics.setup_metrics();
+
+    co_await seastar::async([] {
+        loopback_connection_factory lcf(1);
+        http_server server("test");
```

Comment:
nit, could use 
```c++
auto stop_server = deferred_stop(server);
```

Reply:
I couldn't use this one because the `stop()` method of `http_server` is not `noexcept` so it doesn't satisfy the `stoppable` concept required for `deferred_stop`.